components/SegmentListItem.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+  <li>
+    <div
+      class="information"
+      v-if="date.from || date.to || location">
+      <div
+        class="date"
+        v-if="date.from || date.to">
+        <div
+          class="from"
+          v-if="date.from"
+          v-text="date.from"/>
+        <div
+          class="to"
+          v-if="date.to"
+          v-text="date.to"/>
+      </div>
+      <div
+        class="location"
+        v-if="location"
+        v-text="location"/>
+    </div>
+    <div
+      class="header"
+      v-if="title || position">
+      <h3
+        class="title"
+        v-if="title"
+        v-text="title"/>
+      <h4
+        class="position"
+        v-if="position"
+        v-text="`(${position})`"/>
+    </div>
+    <p
+      class="description"
+      v-if="description"
+      v-html="description"
+      :class="{ justify: justifyDescription }"/>
+  </li>
+</template>
+
+<script>
+export default {
+  name: 'SegmentListItem',
+  props: {
+    title: {
+      default: '',
+      type: String
+    },
+    description: {
+      default: '',
+      type: String
+    },
+    position: {
+      default: '',
+      type: String
+    },
+    location: {
+      default: '',
+      type: String
+    },
+    date: {
+      default: null,
+      type: Object,
+      validator: (date) => {
```

Comment:
I'm not fully sure if your cv generator wants to allow not adding some of the properties, but I'd expect that all the values must be set. So I'd change all your validators to not only check existence of the property, but also of the value. Then you cann remove every v-if (nearly)